Firebase/Auth/Source/FIRAuth.m

```diff
@@ -1486,7 +1487,8 @@ - (void)internalSignInAnonymouslyWithCompletion:(FIRSignupNewUserCallback)comple
  */
 - (void)possiblyPostAuthStateChangeNotification {
   NSString *token = _currentUser.rawAccessToken;
-  if (_lastNotifiedUserToken == token || [_lastNotifiedUserToken isEqualToString:token]) {
+  if (_lastNotifiedUserToken == token ||
```

Comment:
This should probably also be using a string comparison and not a direct pointer comparison. The `nil` check can be moved to the beginning of the conditional.

Reply:
That would change the logic if both were nil. And at least the unit tests depend on not returning when token is nil, but _lastNotification is not.

Reply:
I missed the string equivalence check later on; there's nothing wrong with this code.